migration/wagtail_hooks.py

```diff
@@ -0,0 +1,84 @@
+from django.http import HttpResponseRedirect
+from django.urls import include, path
+from django.utils.translation import gettext as _
+
+from wagtail.core import hooks
+from wagtail.contrib.modeladmin.options import ModelAdmin, ModelAdminGroup, modeladmin_register
+from wagtail.contrib.modeladmin.views import InspectView
+
+from .button_helper import MigrationFailureButtonHelper
+from .models import MigrationFailure
+from .permission_helper import MigrationFailurePermissionHelper
+
+
+class MigrationFailureAdminInspectView(InspectView):
+    def get_context_data(self):
+        try:
+            data = self.instance.data.copy()
+        except AttributeError:
+            data = {}
+
+        return super().get_context_data(**data)
+
+
+class MigrationFailureAdmin(ModelAdmin):
+    model = MigrationFailure
+    button_helper_class = MigrationFailureButtonHelper
+    permission_helper_class = MigrationFailurePermissionHelper
+    # create_view_class = MigrationFailureCreateView
+    inspect_view_enabled = True
+    inspect_view_class = MigrationFailureAdminInspectView
+    menu_label = _('MigrationFailures')
+    menu_icon = 'folder'
+    menu_order = 200
+    add_to_settings_menu = False
+    exclude_from_explorer = False
+
+    list_display = (
+        'action_name',
+        'object_name',
+        'pid',
+        'exception_type',
+    )
+    list_filter = (
+        'action_name',
```

Comment:
Julgo ser adequado se for um choices ou se não houver tantos tipos de `action_name`.